packages/features/form-builder/fieldTypes.ts

```diff
@@ -0,0 +1,149 @@
+import type z from "zod";
+
+import type { FieldType, fieldTypeConfigSchema } from "./schema";
+
+export const fieldTypesConfigMap: Record<FieldType, z.infer<typeof fieldTypeConfigSchema>> = {
+  // This won't be stored in DB. It allows UI to be configured from the codebase for all existing booking fields stored in DB as well
+  // Candidates for this are:
+  // - Anything that you want to show in App UI only.
+  // - Default values that are shown in UI that are supposed to be changed for existing bookingFields as well if user is using default values
+  name: {
+    label: "Name",
+    value: "name",
+    isTextType: true,
+    systemOnly: true,
+    variantsConfig: {
+      toggleLabel: 'Split "Full name" into "First name" and "Last name"',
+      defaultVariant: "fullName",
+      variants: {
+        firstAndLastName: {
+          label: "first_last_name",
+          fieldsMap: {
+            firstName: {
+              defaultLabel: "first_name",
+              canChangeRequirability: false,
+            },
+            lastName: {
+              defaultLabel: "last_name",
+              canChangeRequirability: true,
+            },
+          },
+        },
+        fullName: {
+          label: "your_name",
+          fieldsMap: {
+            fullName: {
+              defaultLabel: "your_name",
+              defaultPlaceholder: "example_name",
+              canChangeRequirability: false,
+            },
+          },
+        },
+      },
+      defaultValue: {
+        variants: {
+          firstAndLastName: {
+            // Configures variant fields
+            // This array form(in comparison to a generic component form) has the benefit that we can allow configuring placeholder, label, required etc. for each variant
+            // Doing this in a generic component form would require a lot of work in terms of supporting variables maybe that would be read by the component.
+            fields: [
+              {
+                // This name won't be configurable by user. User can always configure the main field name
+                name: "firstName",
+                type: "text",
```

Comment:
type and required shouldn't be part of it, they should be using their default values from fieldTypeConfig